jormungandr/src/network/mod.rs

```diff
@@ -381,11 +381,13 @@ fn handle_propagation_msg(
 
 fn start_gossiping(state: GlobalStateR, channels: Channels) -> impl Future<Item = (), Error = ()> {
     let config = &state.config;
-    let topology = state.topology.clone();
+    let topology_accept = state.topology.clone();
```

Comment:
I had to duplicate the topology. I'm not fully aware of what consequences could this has.

Reply:
It's a `Arc<Mutex<...>>` under the hood